src/server/qgsfcgiserverresponse.cpp

```diff
@@ -23,15 +23,141 @@
 #include <fcgi_stdio.h>
 #include <QDebug>
 
+#include <unistd.h>
+#include "qgslogger.h"
+
+#ifdef Q_OS_LINUX
+#include <sys/types.h>
+#include <sys/socket.h>
```

Comment:
Would it be possible to use Qt sockets instead of Unix based sockets?

Reply:
The socket is provided by the fcgi lib and I do not know how to associate a Qt socket to a socket fd. By the way I need to set the SO_LINGER option on the socket and it is not available with QAbstractSocket::SocketOption

Reply:
I never used it but did you take a look to `QLocalSocket::setSocketDescriptor(...)`?

Reply:
But I still need to set a specific option to the socket and I can not do that with the QLocalSocket. But I could set the option with the C lib and then use the QLocalSocket to send signal when the socket is disconnected. Is it what you have in mind?